### PR TITLE
Fix stackoverflow for attribute handler for circular reference

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs [hunk 1]
```diff
@@ -43,40 +44,48 @@ public virtual object Handle(object obj, HandleModelAttributesContext context)
...
+
+                context.NestedType.Pop();
             }
 
-            return obj;
+            return result;
         }
 
-        protected virtual bool ShouldHandle(PropInfo currentPropInfo, object declaringObject, HandleModelAttributesContext context)
+        protected virtual bool ShouldHandle(object currentObj, object declaringObject, PropInfo currentPropInfo, HandleModelAttributesContext context)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/BaseModelAttributeHandler.cs]: >object declaringObject, PropInfo currentPropInfo, [](start = 63, length = 49)

move to context?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 2]
```diff
@@ -52,73 +52,54 @@ public MarkdownContentHandlerImpl(Type type, IModelAttributeHandler handler) : b
...
 
-            protected override void HandleCurrentProperty(object declaringObject, PropertyInfo currentPropertyInfo, HandleModelAttributesContext context)
+            protected override object HandleCurrent(object currentObj, object declaringObject, PropertyInfo currentPropertyInfo, HandleModelAttributesContext context)
             {
-                var obj = currentPropertyInfo.GetValue(declaringObject);
-                if (obj == null)
+                if (currentObj == null && currentPropertyInfo != null && declaringObject != null)
                 {
-                    return;
+                    currentObj = currentPropertyInfo.GetValue(declaringObject);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: >GetValue [](start = 53, length = 8)

perf: too many reflections in code.

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs [hunk 3]
```diff
@@ -52,73 +52,54 @@ public MarkdownContentHandlerImpl(Type type, IModelAttributeHandler handler) : b
...
+                {
+                    return null;
+                }
+
+                var val = currentObj as string;
                 if (val != null)
                 {
                     var marked = Markup(val, context);
-                    currentPropertyInfo.SetValue(declaringObject, marked);
+                    currentPropertyInfo?.SetValue(declaringObject, marked);
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: >currentPropertyInfo?.SetValue [](start = 20, length = 29)

if `currentPropertyInfo` is null, why mark the content?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/HandleModelAttributesContext.cs [hunk 4]
```diff
@@ -11,6 +12,7 @@ namespace Microsoft.DocAsCode.Build.Common
 
     public class HandleModelAttributesContext
     {
+        internal Stack<Type> NestedType { get; set; } = new Stack<Type>();
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/HandleModelAttributesContext.cs]: >NestedType [](start = 29, length = 10)

Can use a counter?

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs (hunk 3 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/Handlers/MarkdownContentHandler.cs]: return `marked`, for case of `List<string>`

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
:shipit:

**Pull request merged**